### PR TITLE
Multi manifest build - arm64/arm (new) + amd64 (current)

### LINKED ISSUE #51: Build docker image as multiplatform image
Currently, the pact-cli can only run on Intel/AMD based architectures. With the advent of Apple's M1 chips and Amazon's Graviton machines, it would be very convenient if we could also run these docker images on those arm based processors. Do you have any plans to release a multiplatform version of the pact-cli?

### PR DESCRIPTION
**Pull request by YOU54F**
closes #51 

create and publish arm64 + arm versions of pact cli and pact broker docker images for use in raspberry Pi's / mac m1/2 etc.

currently publishing to personal docker hub acc for testing

- https://hub.docker.com/r/you54f/pact-cli

Related

- https://github.com/pact-foundation/pact-broker-docker/issues/30
- https://hub.docker.com/r/you54f/pact-broker
- https://github.com/pact-foundation/pact-broker-docker/pull/123

Have spun up the images briefly on an m1 pro macbook but would warrant some decent testing - I have several pi's (3b+ and 4) so will give them a whirl

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM alpine:3.17.2
+FROM alpine:3.17.3
```
Comment (YOU54F) [inline Dockerfile:1]: trivy scan reported vulns

File: Dockerfile [hunk 2]
```diff
@@ -9,19 +9,19 @@ ADD docker/gemrc /root/.gemrc
 ADD docker/pact /usr/local/bin/pact
 
 RUN apk update \
-  && apk add ruby=3.1.3-r0 \
-             ruby-bigdecimal=3.1.3-r0 \
+  && apk add ruby=3.1.4-r0 \
```
Comment (YOU54F) [inline Dockerfile:12]: 3.1.3 not available in apk world

File: script/scan-inside-docker-container.sh [hunk 3]
```diff
@@ -3,5 +3,12 @@ set -eu
 
 # TODO set exit code back to 1
 
-wget -q -O - https://raw.githubusercontent.com/aquasecurity/trivy/master/contrib/install.sh | sh -s -- -b /usr/local/bin
+# TODO - Remove conditional when https://github.com/aquasecurity/trivy/pull/3985 merged
```
Comment (YOU54F) [inline script/scan-inside-docker-container.sh:6]: see https://github.com/aquasecurity/trivy/pull/3985

File: script/scan-inside-docker-container.sh [hunk 4]
```diff
@@ -3,5 +3,12 @@ set -eu
...
+if [[ "$(uname -m)" == 'armv7l' ]]; then
+  # wget -q -O - https://github.com/aquasecurity/trivy/releases/download/v0.39.0/trivy_0.39.0_Linux-ARM.tar.gz | tar xvz -C /usr/local/bin
+  wget -q -O - https://raw.githubusercontent.com/YOU54F/trivy/patch-1/contrib/install.sh | sh -s -- -b /usr/local/bin
```
Comment (mefellows) [inline script/scan-inside-docker-container.sh:9]: 😆

Comment (YOU54F) [reply script/scan-inside-docker-container.sh:9]: Always gotta be shaving some yaks 😅

Comment (YOU54F) [reply script/scan-inside-docker-container.sh]: just dropped this, and only running the scan for the `amd64` image, due to time constraints and they seemed to report the same vulns across all three so far

**Pull request merged**